lib/improver/tests/wind_downscaling/test_FrictionVelocity.py

```diff
@@ -0,0 +1,105 @@
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for plugin wind_downscaling.FrictionVelocity"""
+
+import unittest
+import numpy as np
+
+from iris.tests import IrisTest
+
+from improver.constants import RMDI
+from improver.wind_downscaling import FrictionVelocity
+
+
+class Test_process(IrisTest):
+
+    """Test the creation of friction velocity 2D arrays. Note that in the
+    future, use of the Real Missing Data Indicator (RMDI) constant is due to be
+    deprecated in favour of np.nan"""
+
+    def setUp(self):
+        """Creates wind-speed, height, veg roughness and mask 2D arrays."""
+
+        n_x, n_y = 4, 4  # Set array dimensions.
+
+        # Wind speed=u_href=10m/s  Height=h_ref=20m
+        self.u_href = np.full([n_y, n_x], 10, dtype=float)
+        self.h_ref = np.full([n_y, n_x], 20, dtype=float)
+        # Vegetative roughness = 0.5m
+        self.z_0 = np.full([n_y, n_x], 0.5, dtype=float)
+
+        # Mask for land/sea - True for land-points, false for sea.
+        self.mask = np.full([n_y, n_x], False, dtype=bool)
+        # Mask has 'land' in centre bounded by sea points.
+        self.mask[1:n_y-1, 1:n_x-1] = True
+
+    def test_returns_expected_values(self):
+        """Test that the function returns correct 2D array of floats. """
+
+        # Equation is (K=0.4): ustar = K * (u_href / ln(h_ref / z_0))
+        expected_out = np.array([[RMDI, RMDI, RMDI, RMDI],
+                                 [RMDI, 1.08434, 1.08434, RMDI],
+                                 [RMDI, 1.08434, 1.08434, RMDI],
+                                 [RMDI, RMDI, RMDI, RMDI]])
+
+        result = FrictionVelocity(self.u_href, self.h_ref,
+                                  self.z_0, self.mask).process()
+
+        self.assertIsInstance(result, np.ndarray)
+        self.assertArrayAlmostEqual(result, expected_out)
+
+    def test_handles_nan_values(self):
+        """Test that the function accepts NaN values correctly. """
+
+        self.u_href[1, 1] = np.nan  # Adds NaN value
+
+        expected_out = np.array([[RMDI, RMDI, RMDI, RMDI],
+                                 [RMDI, np.nan, 1.08434, RMDI],
+                                 [RMDI, 1.08434, 1.08434, RMDI],
+                                 [RMDI, RMDI, RMDI, RMDI]])
+
+        result = FrictionVelocity(self.u_href, self.h_ref,
+                                  self.z_0, self.mask).process()
+
+        self.assertIsInstance(result, np.ndarray)
+        self.assertArrayAlmostEqual(result, expected_out)
+
```

Comment:
Added test below to check that input array size is checked and that a `ValueError` is raised if an array is of different size.